agenda-api/src/main/java/org/exoplatform/agenda/service/AgendaEventService.java

```diff
@@ -303,4 +304,16 @@ public Event createEventExceptionalOccurrence(long eventId,
                                                                             AgendaException,
                                                                             ObjectNotFoundException;
 
+  /**
+   * Search the list of events available with query for the currentUser
+   * @param currentUser current user connected
+   * @param query Term to search
+   * @param offset offset
+   * @param limit Limit of events to retrieve
+   * @return
+   */
+  List<EventSearchResultEntity> search(Identity currentUser,
```

Comment:
It's better to not pass and require all the Identity object here, you need just the `identityId` isn't it ?